code/modules/vehicle/vehicle.dm

```diff
@@ -93,25 +134,30 @@
 
 //MOVEMENT
 /obj/vehicle/relaymove(mob/user, direction)
+	if(key_type && !is_key(inserted_key))
+		to_chat(user, "<span class='warning'>[src] has no key inserted!</span>")
+		return
+
 	if(user.incapacitated())
 		unbuckle_mob(user)
 		return
 
-	if(world.time < next_vehicle_move)
+	if(world.time < last_vehicle_move + ((last_move_diagonal? 2 : 1) * (vehicle_move_delay + config.human_delay)))
 		return
-	next_vehicle_move = world.time + vehicle_move_delay
-	if(keycheck(user))
+	last_vehicle_move = world.time
+
+	if(held_keycheck(user))
+		var/turf/next = get_step(src, direction)
 		if(!Process_Spacemove(direction) || !isturf(loc))
 			return
 		step(src, direction)
 
-		if(has_buckled_mobs())
-			for(var/m in buckled_mobs)
-				var/mob/living/buckled_mob = m
-				if(buckled_mob.loc != loc)
-					buckled_mob.buckled = null //Temporary, so Move() succeeds.
-					buckled_mob.buckled = src //Restoring
```

Comment:
This ugly hack, above, isn't needed anymore, since we have a system in place that relays buckled mobs movement when the thing they're buckled to moves---this hack of unbuckling then rebuckling was only necessary because of that.